packages/plugin-http-client/src/HttpClient.ts

```diff
@@ -0,0 +1,160 @@
+import type {
+  Dependencies,
+  HttpAgent,
+  HttpAgentRequestOptions,
+  UnknownParameters,
+} from '@ima/core';
+
+declare module '@ima/core' {
+  interface OCAliasMap {
+    HttpClientDefaultProcessors: Processor[];
+  }
+}
+
+import {
+  AbstractProcessor,
+  Processor,
+  Operation,
+  ProcessorParams,
+} from './AbstractProcessor';
+
+export const OPTION_TRANSFORM_PROCESSORS = 'transformProcessors';
+
+export type HttpClientRequestOptions = {
+  [OPTION_TRANSFORM_PROCESSORS]?: (processor: Processor[]) => Processor[];
+} & HttpAgentRequestOptions;
+
+export type HttpClientRequestMethod =
+  | 'get'
+  | 'put'
+  | 'post'
+  | 'patch'
+  | 'delete';
+
+export type HttpClientRequest = {
+  method: HttpClientRequestMethod;
+  url: string;
+  data?: Record<string, any>;
+  options?: HttpClientRequestOptions;
+};
+
+/**
+ * HttpClient use ima HttpAgent and adds support for processors.
+ */
+export class HttpClient {
+  #http: HttpAgent;
+  #defaultProcessors: Processor[];
+
+  static get $dependencies(): Dependencies {
+    return ['$Http', '...?HttpClientDefaultProcessors'];
+  }
+
+  constructor(http: HttpAgent, ...defaultProcessors: Processor[]) {
+    this.#http = http;
+
+    this.#defaultProcessors = defaultProcessors.filter(
+      item => item instanceof AbstractProcessor
```

Comment:
It could somehow communicate that it filtered out some given processors, if that ever happens. e.g. through console.warn..

Reply:
I changed it  so that represent  why i used that